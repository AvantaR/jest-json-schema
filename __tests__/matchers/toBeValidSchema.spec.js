const toBeValidSchemaUnderTest = require('../../index').matchers.toBeValidSchema;

expect.extend({
  toBeValidSchemaUnderTest,
});

describe('toBeValidSchema', () => {
  let schema;

  beforeEach(() => {
    schema = {
      properties: {
        hello: { type: 'string' },
      },
      required: ['hello'],
    };
  });

  it('does not throw', () => {
    expect(schema).toBeValidSchemaUnderTest();
  });

  it('fails schemas with invalid properties', () => {
    schema.properties = [];
    expect(() => expect(schema).toBeValidSchemaUnderTest())
      .toThrowErrorMatchingSnapshot();
  });

  it('fails schemas of invalid types', () => {
    expect(() => expect('schema').toBeValidSchemaUnderTest())
      .toThrowErrorMatchingSnapshot();
  });

  it('fails for valid schema when using .not', () => {
    expect(() => expect(schema).not.toBeValidSchemaUnderTest())
      .toThrowErrorMatchingSnapshot();
  });

  it('assertion error matcherResult property contains matcher name and actual value', () => {
    schema.properties = [];
    try {
      expect(schema).toBeValidSchemaUnderTest();
    } catch (error) {
      expect(error.matcherResult).toEqual({
        actual: schema,
        message: expect.any(Function),
        name: 'toBeValidSchema',
        pass: false,
      });
    }
  });
});
