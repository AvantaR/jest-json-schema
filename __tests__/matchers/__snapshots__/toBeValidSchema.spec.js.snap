// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toBeValidSchema fails for valid schema when using .not 1`] = `
"[2mexpect([22m[31mreceived[39m[2m).not.toBeValidSchema([22m[2m)[22m

Expected input not to be a valid JSON schema"
`;

exports[`toBeValidSchema fails schemas of invalid types 1`] = `
"[2mexpect([22m[31mschema[39m[2m).toBeValidSchema([22m[2m)[22m

schema
[31m  should be object
[39m"
`;

exports[`toBeValidSchema fails schemas with invalid properties 1`] = `
"[2mexpect([22m[31mschema[39m[2m).toBeValidSchema([22m[2m)[22m

schema
[31m  .properties should be object
[39m"
`;
